feat: tag assembly task

ingarabr opened the pull request:
Add a tag for the assembly tasks. This can be used to restrict resource usages.

**Why do we need this?**

Version 2 of sbt-assembly uses more resources. This is problematic when executing `assembly` on a multi-module project causing different types of errors. I've encountered memory and to many open files issue. Adding a tag gives the user of the plugin the option to restrict concurrent executions with `Global / concurrentRestrictions += Tags.limit(Assembly.assemblyTag, 1)`

**Things to consider**
- Should we reuse some of the existing tags? Sbt has defined both cpu, memory and disk tags.
- Should the plugin have a default limit on the tag?

ingarabr committed:
feat: tag assemly task

Add a tag for the assembly tasks. This can be used to restrict
resource usages.

eed3si9n reviewed (approved):
Thanks for the contribution!

eed3si9n commented:
> Should the plugin have a default limit on the tag?

I think it's a good idea to put default limit on 1.

ingarabr committed:
fix: configure assembly task concurrent restriction

This to prevent the task to fail due to resource starvation.
For instance cpu and to many open files. The issue will most
likely occur on multi-module project where we do assembly on
more than one project. Setting the default to one will ensure
that it will work and can be overridden if needed.

ingarabr commented:
I agree that a default limit of 1 can be a good default. Please take a look at  a04849b
